.github/workflows/reusable-build-docs.yaml

```
@@ -0,0 +1,120 @@
+---
+name: Build Ansible package docs
+
+"on":
+  workflow_call:
+    inputs:
+      repository-owner:
+        description: GitHub account or org that owns the repository
+        type: string
+        required: false
+        default: ansible
+      repository-name:
+        description: Name of the GitHub repository
+        type: string
+        required: false
+        default: ansible-documentation
+      repository-branch:
+        description: Branch, tag, or commit SHA
+        type: string
+        required: false
+        default: devel
+      ansible-package-version:
+        type: string
+        description: Ansible community package version
+        required: false
+        default: devel
+    secrets:
+      DOCS_BOT_TOKEN:
+        required: true
+
+env:
+  PACKAGE_VERSION: ${{ inputs.ansible-package-version }}
+
+jobs:
+  build-package-docs:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Ansible documentation
+      uses: actions/checkout@v4
+      with:
+        repository: >-
+          ${{
+            inputs.repository-owner
+          }}/${{
+            inputs.repository-name
+          }}
+        ref: ${{ inputs.repository-branch }}
+        path: build-directory
+
+    - name: Setup nox
+      uses: wntrblm/nox@2024.10.09
+
+    - name: Output Python info
+      run: python --version --version && which python
+
+    - name: Graft ansible-core
+      run: nox -s clone-core
+      working-directory: build-directory
+
+    - name: Install project requirements
+      run: >-
+        python -m pip install
+        -r tests/requirements.in
+        -c tests/requirements.txt
+      working-directory: build-directory
+
+    - name: Set the COLLECTION_LIST variable
+      if: env.PACKAGE_VERSION != 'devel'
+      run: >-
+        echo COLLECTION_LIST="${PACKAGE_VERSION}"
+        >> "${GITHUB_ENV}"
+
+    - name: Set the VERSION variable
+      run: echo VERSION="${PACKAGE_VERSION}" >> "${GITHUB_ENV}"
+
+    - name: Build the Ansible community package docs
+      run: make webdocs ANSIBLE_VERSION="${COLLECTION_LIST}"
```

Alternatively, remove the `COLLECTION_LIST` task above and use this here instead:
```suggestion
      run: make webdocs ANSIBLE_VERSION="${env.PACKAGE_VERSION != 'devel' && env.PACKAGE_VERSION || '' }"
```

---

Nice one @felixfontein Let's go with this!